### PR TITLE
Remove type constraint to permit field b.c. on curl operator

### PR DESCRIPTION
**Pull request by akshaysridhar**
Currently, divergence and gradient operators support field b.c., curl does not due to a type constraint in the `finitedifference` boundary condition types. This PR removes the constraint and tests that the method works when the supplied boundary condition is a field. 
e.g. For reference, assigning a field b.c `fω¹_bc` where `fω¹_bc` is a Contravariant2Vector Field returns the following error. 
```
bottom = Operators.SetCurl(fω¹_bc)
ERROR: MethodError: no method matching ClimaCore.Operators.SetCurl(::ClimaCore.Fields.Field{ClimaCore.DataLayouts.IFH{Contravariant2Vector{Float64}, 5, SubArray{Float64, 3, Array{Float64, 4}, Tuple{Int64, Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}, Base.Slice{Base.OneTo{Int64}}}, true}}, ...
```
- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by simonbyrne — APPROVED** [review]

File: test/Operators/hybrid/2d.jl [hunk 1]
```diff
@@ -547,6 +547,56 @@ end
...
 end
 
+@testset "curl (field b.c.)" begin
+    hv_center_space, hv_face_space =
+        hvspace_2D(xlim = (-pi, pi), zlim = (-pi, pi))
+
+    ccoords = Fields.coordinate_field(hv_center_space)
+    fcoords = Fields.coordinate_field(hv_face_space)
+    fcoords_1 = Fields.level(fcoords, ClimaCore.Utilities.half)
+    curl_bcfield = Geometry.Covariant2Vector.(0.0 .* fcoords_1)
```
Comment (simonbyrne) [inline test/Operators/hybrid/2d.jl:557]: You need to specify a component of `fcoords_1` (`x` or `z`)?
```suggestion
    curl_bcfield = Geometry.Covariant2Vector.(0.0 .* fcoords_1.x)
```

File: test/Operators/hybrid/2d.jl [hunk 2]
```diff
@@ -547,6 +547,56 @@ end
...
 end
 
+@testset "curl (field b.c.)" begin
+    hv_center_space, hv_face_space =
+        hvspace_2D(xlim = (-pi, pi), zlim = (-pi, pi))
+
+    ccoords = Fields.coordinate_field(hv_center_space)
+    fcoords = Fields.coordinate_field(hv_face_space)
+    fcoords_1 = Fields.level(fcoords, ClimaCore.Utilities.half)
+    curl_bcfield = Geometry.Covariant2Vector.(0.0 .* fcoords_1.z)
```
Comment (simonbyrne) [inline test/Operators/hybrid/2d.jl:557]: Should this be?
```suggestion
    curl_bcfield = Geometry.Covariant1Vector.(0.0 .* fcoords_1.z)
```

Comment (simonbyrne) [reply test/Operators/hybrid/2d.jl:557]: @akshaysridhar ?

**Comment by sriharshakandala** [comment]
LGTM

**Review by sriharshakandala — APPROVED** [review]
